kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/Retry.kt

```diff
@@ -85,6 +85,7 @@ suspend fun <T> retry(
          // else ignore and continue
       }
       retrySoFar++
+      if (retrySoFar >= config.maxRetry) break
```

Comment:
Why only checking `retrySoFar`? It could also be that the end time has passed, even if fewer than max retries have been taken.

Reply:
Also note the failure message is misleading: `Test failed after ${config.delay.toLong(DurationUnit.SECONDS)} seconds`. Here, `delay` is only the initial delay. It ignores the multiplier, and it ignores how many times we've delayed.

Reply:
@klitoskyriacou 

Thank you for review. I've revised code to reflect your code review.